install-dev/upgrade/upgrade.php

```diff
@@ -110,7 +110,8 @@
 }
 
 $result = '<?xml version="1.0" encoding="UTF-8"?>';
-if (empty($upgrade->hasFailure())) {
+$hasFailure = $upgrade->hasFailure();
```

Comment:
Actually, when we look at this function, we can find:

```php
        public function hasFailure()
        {
            return !empty($this->failureList);
        }
```

I don't understand why we have an `empty()` at the first place.
```php
if (!$upgrade->hasFailure()) {
```
was enough.